Index: unit.cpp
===================================================================
RCS file: /home/cvs/atlantis/unit.cpp,v
retrieving revision 1.39
diff -u -r1.39 unit.cpp
--- unit.cpp	2001/12/02 22:19:45	1.39
+++ unit.cpp	2001/12/15 09:20:46
@@ -1269,7 +1269,7 @@
 	int amulets = items.GetNum(I_AMULETOFTS);
 	if((u->items.GetNum(I_RINGOFI) < 1) || (amulets < 1)) return 0;
 	int men = GetMen();
-	if(men < amulets) return 1;
+	if(men <= amulets) return 1;
 	if(!Globals->PROPORTIONAL_AMTS_USAGE) return 0;
 	if(getrandom(men) < amulets) return 1;
 	return 0;
