Index: army.cpp
===================================================================
RCS file: /home/cvs/atlantis/army.cpp,v
retrieving revision 1.34
retrieving revision 1.34.2.1
diff -u -r1.34 -r1.34.2.1
--- army.cpp	2001/11/24 21:54:08	1.34
+++ army.cpp	2001/12/10 08:06:21	1.34.2.1
@@ -1027,15 +1027,17 @@
 		/* 4.1 Check whether defense is allowed against this weapon */
 		if((flags & WeaponType::NODEFENSE) && (tlev > 0)) tlev = 0;
 
-		/* 4.2 Check relative weapon length */
-		int attLen = 1;
-		int defLen = 1;
-		if(flags & WeaponType::LONG) attLen = 2;
-		else if(flags & WeaponType::SHORT) attLen = 0;
-		if(tarFlags & WeaponType::LONG) defLen = 2;
-		else if(tarFlags & WeaponType::SHORT) defLen = 0;
-		if(attLen > defLen) attackLevel++;
-		else if(defLen > attLen) tlev++;
+		if(!(flags & WeaponType::RANGED)) {
+			/* 4.2 Check relative weapon length */
+			int attLen = 1;
+			int defLen = 1;
+			if(flags & WeaponType::LONG) attLen = 2;
+			else if(flags & WeaponType::SHORT) attLen = 0;
+			if(tarFlags & WeaponType::LONG) defLen = 2;
+			else if(tarFlags & WeaponType::SHORT) defLen = 0;
+			if(attLen > defLen) attackLevel++;
+			else if(defLen > attLen) tlev++;
+		}
 
 		/* 4.3 Add bonuses versus mounted */
 		if(tar->riding != -1) attackLevel += mountBonus;
